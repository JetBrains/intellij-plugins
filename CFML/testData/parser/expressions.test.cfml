<cfscript>
  var arr = [];
  var obj = {};

  public array function getOptions() {
    return (isNull(variables.options)) ? [] : variables.options;
  }

  public void function setOptions(array options = []) {

  }

  public void function setOptions(struct options = {}) {

  }

  someComponent.somefunction(somearg = 1, test = { "somekey" = "somevalue" });

  public void function addRoute( any routes, string target, any methods = [ ], string statusCode = '' ) {
  }
</cfscript>