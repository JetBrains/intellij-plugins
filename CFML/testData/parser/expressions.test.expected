CfmlFile:expressions.cfml
  Tag
    PsiElement(OPENER)('<')
    PsiElement(CF_TAG_NAME)('cfscript')
    PsiElement(R_ANGLEBRACKET)('>')
    Assignment
      PsiElement(VAR_KEYWORD)('var')
      ReferenceExpression
        PsiElement(IDENTIFIER)('arr')
      PsiElement(ASSIGN)('=')
      PsiElement(L_SQUAREBRACKET)('[')
      PsiElement(R_SQUAREBRACKET)(']')
    PsiElement(SEMICOLON)(';')
    Assignment
      PsiElement(VAR_KEYWORD)('var')
      ReferenceExpression
        PsiElement(IDENTIFIER)('obj')
      PsiElement(ASSIGN)('=')
      PsiElement(L_CURLYBRACKET)('{')
      PsiElement(R_CURLYBRACKET)('}')
    PsiElement(SEMICOLON)(';')
    FunctionDefinition
      PsiElement(PUBLIC_KEYWORD)('public')
      TYPE
        PsiElement(ARRAY_TYPE)('array')
      PsiElement(FUNCTION_KEYWORD)('function')
      PsiElement(IDENTIFIER)('getOptions')
      ParametersList
        PsiElement(L_BRACKET)('(')
        PsiElement(R_BRACKET)(')')
      FUNCTIONBODY
        BLOCK_OF_STATEMENTS
          PsiElement(L_CURLYBRACKET)('{')
          PsiElement(RETURN_KEYWORD)('return')
          TernaryExpression
            PsiElement(L_BRACKET)('(')
            FunctionCallExpression
              ReferenceExpression
                PsiElement(IDENTIFIER)('isNull')
              ArgumentList
                PsiElement(L_BRACKET)('(')
                ReferenceExpression
                  PsiElement(SCOPE_KEYWORD)('variables')
                  PsiElement(POINT)('.')
                  PsiElement(IDENTIFIER)('options')
                PsiElement(R_BRACKET)(')')
            PsiElement(R_BRACKET)(')')
            PsiElement(QUESTION)('?')
            PsiElement(L_SQUAREBRACKET)('[')
            PsiElement(R_SQUAREBRACKET)(']')
            PsiElement(DOTDOT)(':')
            ReferenceExpression
              PsiElement(SCOPE_KEYWORD)('variables')
              PsiElement(POINT)('.')
              PsiElement(IDENTIFIER)('options')
          PsiElement(SEMICOLON)(';')
          PsiElement(R_CURLYBRACKET)('}')
    FunctionDefinition
      PsiElement(PUBLIC_KEYWORD)('public')
      TYPE
        PsiElement(VOID_TYPE)('void')
      PsiElement(FUNCTION_KEYWORD)('function')
      PsiElement(IDENTIFIER)('setOptions')
      ParametersList
        PsiElement(L_BRACKET)('(')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(ARRAY_TYPE)('array')
          PsiElement(IDENTIFIER)('options')
          PsiElement(ASSIGN)('=')
          VALUE
            PsiElement(L_SQUAREBRACKET)('[')
            PsiElement(R_SQUAREBRACKET)(']')
        PsiElement(R_BRACKET)(')')
      FUNCTIONBODY
        BLOCK_OF_STATEMENTS
          PsiElement(L_CURLYBRACKET)('{')
          PsiElement(R_CURLYBRACKET)('}')
    FunctionDefinition
      PsiElement(PUBLIC_KEYWORD)('public')
      TYPE
        PsiElement(VOID_TYPE)('void')
      PsiElement(FUNCTION_KEYWORD)('function')
      PsiElement(IDENTIFIER)('setOptions')
      ParametersList
        PsiElement(L_BRACKET)('(')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(STRUCT_TYPE)('struct')
          PsiElement(IDENTIFIER)('options')
          PsiElement(ASSIGN)('=')
          VALUE
            PsiElement(L_CURLYBRACKET)('{')
            PsiElement(R_CURLYBRACKET)('}')
        PsiElement(R_BRACKET)(')')
      FUNCTIONBODY
        BLOCK_OF_STATEMENTS
          PsiElement(L_CURLYBRACKET)('{')
          PsiElement(R_CURLYBRACKET)('}')
    FunctionCallExpression
      ReferenceExpression
        ReferenceExpression
          PsiElement(IDENTIFIER)('someComponent')
        PsiElement(POINT)('.')
        PsiElement(IDENTIFIER)('somefunction')
      ArgumentList
        PsiElement(L_BRACKET)('(')
        Assignment
          Argument somearg
            PsiElement(IDENTIFIER)('somearg')
          PsiElement(ASSIGN)('=')
          IntegerLiteral
            PsiElement(INTEGER)('1')
        PsiElement(COMMA)(',')
        Assignment
          Argument test
            PsiElement(IDENTIFIER)('test')
          PsiElement(ASSIGN)('=')
          PsiElement(L_CURLYBRACKET)('{')
          Assignment
            PsiElement(DOUBLE_QUOTE)('"')
            PsiElement(STRING_TEXT)('somekey')
            PsiElement(DOUBLE_QUOTE_CLOSER)('"')
            PsiElement(ASSIGN)('=')
            StringLiteral
              PsiElement(DOUBLE_QUOTE)('"')
              PsiElement(STRING_TEXT)('somevalue')
              PsiElement(DOUBLE_QUOTE_CLOSER)('"')
          PsiElement(R_CURLYBRACKET)('}')
        PsiElement(R_BRACKET)(')')
    PsiElement(SEMICOLON)(';')
    FunctionDefinition
      PsiElement(PUBLIC_KEYWORD)('public')
      TYPE
        PsiElement(VOID_TYPE)('void')
      PsiElement(FUNCTION_KEYWORD)('function')
      PsiElement(IDENTIFIER)('addRoute')
      ParametersList
        PsiElement(L_BRACKET)('(')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(ANY_TYPE)('any')
          PsiElement(IDENTIFIER)('routes')
        PsiElement(COMMA)(',')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(STRING_TYPE)('string')
          PsiElement(IDENTIFIER)('target')
        PsiElement(COMMA)(',')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(ANY_TYPE)('any')
          PsiElement(IDENTIFIER)('methods')
          PsiElement(ASSIGN)('=')
          VALUE
            PsiElement(L_SQUAREBRACKET)('[')
            PsiElement(R_SQUAREBRACKET)(']')
        PsiElement(COMMA)(',')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(STRING_TYPE)('string')
          PsiElement(IDENTIFIER)('statusCode')
          PsiElement(ASSIGN)('=')
          VALUE
            PsiElement(SINGLE_QUOTE)(''')
            PsiElement(SINGLE_QUOTE_CLOSER)(''')
        PsiElement(R_BRACKET)(')')
      FUNCTIONBODY
        BLOCK_OF_STATEMENTS
          PsiElement(L_CURLYBRACKET)('{')
          PsiElement(R_CURLYBRACKET)('}')
    PsiElement(LSLASH_ANGLEBRACKET)('</')
    PsiElement(CF_TAG_NAME)('cfscript')
    PsiElement(CLOSER)('>')